### PR TITLE
Differentiate invalid credentials error from unknown errors

### PR DESCRIPTION
**Pull request by Devessier**
Closes https://github.com/AdonisEnProvence/MusicRoom/issues/498

### CODE REVIEW

File: packages/client/__tests__/Authentication.test.tsx [hunk 1]
```diff
@@ -458,31 +458,71 @@ const authenticationModelMachine =
...
+                                                }: TestingContext) => {
+                                                    invariant(
+                                                        screen !== undefined,
+                                                        'Screen must have been rendered',
+                                                    );
+
+                                                    await waitFor(() => {
+                                                        const serverErrorAlert =
+                                                            within(
+                                                                screen.getByTestId(
```
Comment (prastoin) [inline packages/client/__tests__/Authentication.test.tsx:508]: Not important at all but on the sign up screen we do display a toast error, maybe we should pick at least one common solution for both screens ?
Might be later

**Review by prastoin — APPROVED** [review]
Great shaping !
Ready to be merged

File: packages/client/__tests__/Authentication.test.tsx (hunk 1 above)
Comment (Devessier) [reply packages/client/__tests__/Authentication.test.tsx:508]: Indeed, we will have to normalize the UX of the authentication screens.
I did not want to take these decisions alone so I went to the simplest solution, according to the code I already had.

Comment (prastoin) [reply packages/client/__tests__/Authentication.test.tsx:508]: We will then determine all of this details later during our UI/UX improvements journey

**Pull request merged**